File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

let messageList;

document.addEventListener('turbolinks:load', () => {
  messageList = document.getElementById('messageList')

  if (messageList === null) {
    return
  }
  consumer.subscriptions.create("RoomChannel", {
    connected() {
      // Called when the subscription is ready for use on the server
    },
  
    disconnected() {
      // Called when the subscription has been terminated by the server
    },
  
    received(data) {
      // Called when there's incoming data on the websocket for this channel
      messageList.insertAdjacentHTML('beforeend', data['message'])
    }
  });
});
// speak: function() {
//   return this.perform('speak');
// }

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
    # stream_from "some_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    # binding.pry
    Message.create(content: data['message'])
    ApplicationCable.server.broadcast 'room_channel', data['message']
  end
end
